bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/FileInfoBackingCache.java

```diff
@@ -52,7 +52,8 @@ CachedFileInfo loadFileInfo(long ledgerId, byte[] masterKey) throws IOException
                 // have been able to get a reference to it here.
                 // The caller of loadFileInfo owns the refence, and is
                 // responsible for calling the corresponding #release().
-                assert(fi.tryRetain());
+                boolean retained = fi.tryRetain();
+                assert(retained);
```

Comment:
if it is important that retained is true we should check it explicitly and crash, even if assertions are disabled

Reply:
In this case, retained can only be false if someone screws up some code in this file. Noone else has a reference to fi at this point.